### PR TITLE
Update AP contracts and add custom data for purchase

### PR DESCRIPTION
**Pull request by wejohnso673**
- Update AP contracts and endpoints
- Add property bag for custom data to Purchase event

### CODE REVIEW

File: src/Web/Controllers/BasketController.cs [hunk 1]
```diff
@@ -401,6 +401,22 @@ private Purchase SetupPurchase(CheckoutDetailsViewModel checkoutDetails, BasketV
                 BillingAddress = billingAddress,
             };
 
+            Product mostExpensiveProduct = productList.FirstOrDefault();
```
Comment (chadsowald) [inline src/Web/Controllers/BasketController.cs:404]: Could simplify

```var mostExpensiveProduct = productList.Max(p => p.SalesPrice / p.Quantity);```

File: src/ApplicationCore/Entities/FraudProtectionApiModels/AccountProtection/Signup.cs [hunk 2]
```diff
@@ -34,5 +36,8 @@ public class EventMetadataAccountCreate : EventMetadata
         public string SignUpId { get; set; }
 
         public DateTimeOffset CustomerLocalDate { get; set; }
+
+        [JsonConverter(typeof(StringEnumConverter))]
+        public AssessmentType AssessmentType { get; set; }
```
Comment (chadsowald) [inline src/ApplicationCore/Entities/FraudProtectionApiModels/AccountProtection/Signup.cs:41]: :-)

File: src/Web/Controllers/AccountController.cs [hunk 3]
```diff
@@ -117,18 +118,19 @@ private async Task<IActionResult> SignInUser(LoginViewModel model, string return
                 {
                     UserType = AccountProtection.UserType.Consumer,
                     Username = model.Email,
-                    PasswordHash = hashedPassword
+                    UserId = model.Email
```
Comment (chadsowald) [inline src/Web/Controllers/AccountController.cs:121]: Does setting this UserId have any implications?  e.g. will it create a User node in the graph or anything weird even if the SignIn is rejected?  What if the SignIn succeeds?  Does it create a user node in either case?

Comment (wejohnso673) [reply src/Web/Controllers/AccountController.cs:121]: UserId was changed to be a required property.  It is anonymized and sent downstream, but I'm not aware of any implications beyond that.

Regarding graph nodes, I'm not sure how this is working in the new AP world.  The calls aren't going through KG anymore, which would ordinarily be the service responsible for that.

File: src/Web/Controllers/BasketController.cs (hunk 1 above)
Comment (wejohnso673) [reply src/Web/Controllers/BasketController.cs:404]: tried that, but that gives sales price of the most expensive product.  wanted to include the name of the product so we could test sending a string value in custom data.

File: src/ApplicationCore/Entities/FraudProtectionApiModels/AccountProtection/Signup.cs (hunk 2 above)
Comment (wejohnso673) [reply src/ApplicationCore/Entities/FraudProtectionApiModels/AccountProtection/Signup.cs:41]: yes, it was good to see this among the AP contract changes

**Pull request merged**